src/main/java/com/google/songgame/servlets/RoomServlet.java

```diff
@@ -1,25 +1,59 @@
 package com.google.songgame.servlets;
 
 import com.google.gson.Gson;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.stream.Collectors;
+import java.util.Collections;
 
 import org.apache.hc.core5.http.ParseException;
 import com.google.gson.Gson; 
-import java.util.ArrayList;
+import java.util.*;
 
 @WebServlet("/room")
 public final class RoomServlet extends HttpServlet {
 
+  private Gson gson;
+
+  @Override
+  public void init() {
+    gson = new Gson();
+  }
+
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get user name
-    String name = request.getParameter("user-name");
-    response.setContentType("text/html;");
-    response.getWriter().println("Name: " + name);
+    
+    Map<String, String> userJson = readJSONFromRequest(request);
+    List<String> values =  new ArrayList<String>();
+
+    for (String key : userJson.keySet()) {
+      values.add(userJson.get(key));
+    }
+
+    // Save player username and userId to datastore.
+    Entity userEntity = new Entity("User");
```

Comment:
Possibly for the future, consider setting a "key / indentifier" for the entity to be the same as the one generated for the Cookie. That makes it easier to retrieve the Entity later on. You can read more about this here:
https://cloud.google.com/appengine/docs/standard/java/datastore/creating-entity-keys#Kinds_and_identifiers

Reply:
@salnad thanks! If I wait to fix it do you think It'll be a problem?

Reply:
Yeah I think we can wait